### DISCUSSION TITLE
.pnpmfile.cjs changes get overwritten

### DISCUSSION THREAD

**Discussion by roni-frantchi** [renovatebot/renovate#11250] [Request Help] [answered]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Please select which platform you are using if self-hosting.

github.com

### Describe the bug

Steps to reproduce:
1. In a pnpm repository with `pnpm-lock.yaml`, place a `.pnpmfile.cjs` that makes modifies dependencies, like the one in the [official pnpm example](https://pnpm.io/pnpmfile#hooksreadpackagepkg-context-pkg) or this one which is more trivial:
```
function readPackage(pkg, context) {
    pkg.dependencies = {
      ...pkg.dependencies,
      bar: '^2.0.0'
    }
  
  return pkg
}

module.exports = {
  hooks: {
    readPackage
  }
}
```
2. Run `pnpm install` commit and push changes

Expected behavior:  
When Renovate performs lock file maintenance on PRs opened, modifications made by the the pnpmfile should be kept.

Actual behavior:  
While Renovate updates other dependencies, it completely removes all dependencies added by the above file.  

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
This is by design, as we disable scripts by default. So your only options is to use a self-hosted renovate and enable scripts

https://github.com/renovatebot/renovate/blob/9ca604b3453ca55795bd10d3fcad4475c3c8c7cb/lib/manager/npm/post-update/pnpm.ts#L52-L56

https://pnpm.io/pnpmfile#ignore-pnpmfile

**Reply by roni-frantchi** [reply]
Thank you @viceice and apologies on opening this as an issue - I was not aware that was by design.  

May I ask why was that the design that was chosen?  
What are the concerns of not disabling it?

**Reply by rarkins** [reply]
We disable all scripts we can to minimise the attack surface.

**Reply by batbattur** [reply]
Hello guys, I have a similar case where we are currently running renovate on Mend renovate hosted so we need to switch to self-hosted to get `allowScripts` and `pnpmfile.cjs` working.

Question: When switching from Mend Renovate hosted app on github.com to self-hosted, can we uninstall Renovate app in `Installed Github Apps` and revoke its permissions (since we're going to use the self-hosted GHA)? Will there be any complications if we run it self-hosted while also having it run on Mend Renovate hosted app?

**Reply by rarkins** [reply]
You should not run two bots on the same repo

[a comment by an automated account is not shown]
